index.js

```diff
@@ -2171,6 +2179,38 @@
       [TypeRep, $.String, $.Array(a), $.Array($Maybe(b))],
       function(type, key, xs) { return R.map(get(type, key), xs); });
 
+  //# reduce :: Foldable f => (a -> b -> a) -> a -> f b -> a
+  //.
+  //. Takes a binary function, an initial value, and a [Foldable][], and
+  //. applies the function to the initial value and the Foldable's first
+  //. value, then applies the function to the result of the previous
+  //. application and the Foldable's second value. Repeats this process
+  //. until each of the Foldable's values has been used. Returns the initial
+  //. value if the Foldable is empty; the result of the final application
+  //. otherwise.
+  //.
+  //. ```javascript
+  //. > S.reduce(S.add, 0, [1, 2, 3, 4, 5])
+  //. 15
+  //.
+  //. > S.reduce((xs, x) => [x].concat(xs), [], [1, 2, 3, 4, 5])
+  //. [5, 4, 3, 2, 1]
+  //. ```
+  S.reduce =
+  def('reduce',
+      {b: [Foldable]},
+      [$.Function, a, b, a],
+      function(f, acc, foldable) {
+        if (_type(foldable) === 'Array') {
+          for (var idx = 0; idx < foldable.length; idx += 1) {
+            acc = f(acc, foldable[idx]);
```

Comment:
I'd prefer not to reassign to a parameter. Let's rename the parameter `initial` and add `var acc = initial;` before the loop.